service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderControllerTest.java

```diff
@@ -39,25 +49,62 @@
 
     @Test
     void shouldReturnErrorsWhenTheDateOfIssueIsInFuture() {
-        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now().plusDays(1), NAMED_DATE);
+        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now().plusDays(1),
+            NAMED_DATE, ORDER_TYPE_AND_DOCUMENT, time.now());
         final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
         assertThat(callbackResponse.getErrors()).containsOnlyOnce("Date of issue cannot be in the future");
     }
 
     @Test
     void shouldNotReturnErrorsWhenDateOfIssueIsToday() {
-        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now(), NAMED_DATE);
+        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now(),
+            NAMED_DATE, ORDER_TYPE_AND_DOCUMENT, time.now());
         final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
         assertThat(callbackResponse.getErrors()).isEmpty();
     }
 
     @Test
     void shouldNotReturnErrorsWhenDateOfIssueIsInPast() {
-        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now().minusDays(1), NAMED_DATE);
+        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now().minusDays(1),
+            NAMED_DATE, ORDER_TYPE_AND_DOCUMENT, time.now());
+        final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+        assertThat(callbackResponse.getErrors()).isEmpty();
+    }
+
+    @Test
+    void shouldNotReturnErrorsWhenInterimOrderSubtypeAndInterimEndDateInTheFuture() {
+        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now(),
+            NAMED_DATE, INTERIM_ORDER_TYPE_AND_DOCUMENT, time.now().plusDays(1));
         final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
         assertThat(callbackResponse.getErrors()).isEmpty();
     }
 
+    @Test
+    void shouldNotReturnErrorsWhenInterimOrderSubtypeAndInterimEndDateNotSet() {
+        final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(CaseDetails.builder()
```

Comment:
If you want you can just create a case data object here, assign the relevant fields and pass that to the post mid event method which will do all the necessary conversions for you

Reply:
Also any reason for not using the method you use in the other tests?